engine/src/witness/arb.rs

```diff
@@ -205,15 +199,17 @@ impl super::evm::vault::IngressCallBuilder for ArbCallBuilder {
 				deposit_metadata,
 				tx_hash,
 				deposit_details: Box::new(DepositDetails { tx_hashes: Some(vec![tx_hash.into()]) }),
-				// Defaulting to no broker fees until PRO-1743 is completed.
-				broker_fee: cf_primitives::Beneficiary {
-					account: sp_runtime::AccountId32::new([0; 32]),
-					bps: 0,
-				},
-				affiliate_fees: Default::default(),
-				boost_fee: boost_fee.unwrap_or_default(),
-				dca_params,
-				refund_params: refund_params.map(Box::new),
+				broker_fee: vault_swap_parameters.broker_fee,
+				affiliate_fees: vault_swap_parameters
+					.affiliate_fees
+					.into_iter()
+					.map(|entry| Beneficiary { account: entry.affiliate.into(), bps: entry.fee.into() })
+					.collect_vec()
+					.try_into()
+					.expect("runtime supports at least as many affiliates as we allow in cf_parameters encoding"),
+				boost_fee: vault_swap_parameters.boost_fee.unwrap_or_default(),
+				dca_params: vault_swap_parameters.dca_params,
+				refund_params: Some(Box::new(vault_swap_parameters.refund_params)),
```

Comment:
How come refund params are optional on the extrinsic but we never use None?

Reply:
I guess it's because we wanted to have some flexibility as we've been going back and forth between making this mandatory or not. I think we could consider removing the option since we are now enforcing it in Vault swaps. Wdyt? @dandanlen @msgmaxim

Reply:
Yeah, let's remove `Option`

Reply:
Done